crates/types/src/blockchain/block.rs

```diff
@@ -292,6 +298,44 @@ impl From<Block> for PartialFuelBlock {
                 },
                 transactions,
             },
+            #[cfg(feature = "fault-proving")]
+            Block::V1(BlockV1 {
+                header:
+                    BlockHeader::V2(crate::blockchain::header::BlockHeaderV2 {
```

Comment:
This pattern match is pretty big. Would it be easier to read if you just used the getter traits here, and didn't bother matching on the the `BlockHeader` variant?

I might be missing something that needs special attention, but that's kinda the point of my comment :)